### PR TITLE
[SYCL][ESIMD] Fix LowerESIMD crash on a scalar fptoui.

### PR DESCRIPTION
**Pull request by kbobrovs**
Fix small bug in LowerESIMD.cpp

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/LowerESIMD.cpp [hunk 1]
```diff
@@ -1210,10 +1210,11 @@ PreservedAnalyses SYCLLowerESIMDPass::run(Function &F,
     if (auto CastOp = dyn_cast<llvm::CastInst>(&I)) {
       llvm::Type *DstTy = CastOp->getDestTy();
       auto CastOpcode = CastOp->getOpcode();
-      if ((CastOpcode == llvm::Instruction::FPToUI &&
-           DstTy->getScalarType()->getPrimitiveSizeInBits() <= 32) ||
+      if (isa<FixedVectorType>(DstTy) &&
```
Comment (kychendev) [inline llvm/lib/SYCLLowerIR/LowerESIMD.cpp:1213]: Why not just check for isVectorTy()? Does front-end generate ScalableVectorType?

**Comment by bader** [comment]
@kbobrovs, could you fix `clang-format-check` failure and reply to @kychendev questions, please?

**Comment by bader** [comment]
ping

File: llvm/lib/SYCLLowerIR/LowerESIMD.cpp (hunk 1 above)
Comment (kbobrovs) [reply llvm/lib/SYCLLowerIR/LowerESIMD.cpp:1213]: > Why not just check for isVectorTy()?

VectorType llvm type has been split into VectorType and FixedVectorType type for known vector widths throughout LLVM some time ago. gpu::simd is always a FixedVectorType.

> Does front-end generate ScalableVectorType?

I'm not sure. How is this relevant - can you please elaborate?

**Comment by kbobrovs** [comment]
> @kbobrovs, could you fix clang-format-check failure and reply to @kychendev questions, please?

thanks for reminding; finally done

File: llvm/lib/SYCLLowerIR/LowerESIMD.cpp (hunk 1 above)
Comment (kychendev) [reply llvm/lib/SYCLLowerIR/LowerESIMD.cpp:1242]: ok, so it's always FixedVectorType. I was just wondering if ScalableVectorType is supported.

**Review by kychendev — APPROVED** [review]

**Comment by romanovvlad** [comment]
@kbobrovs  Could you please provide description of the problem for the commit message?

**Comment by kbobrovs** [comment]
@romanovvlad, the problem is LowerESIMD crash on the scalar fptoui LLVM instruction encountered within the kernel. I thought current message reflected that. Maybe "[SYCL][ESIMD] Fix LowerESIMD crash on a scalar fptoui LLVM instr" for extra clarity then - ?

**Pull request merged**